x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -103,19 +107,35 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}
 
 	awsConfig.Region = moduleConfig.DefaultRegion
-
 	svcEC2 := ec2.New(awsConfig)
 	regionsList, err := getRegions(svcEC2)
 	if err != nil {
 		err = errors.Wrap(err, "getRegions failed")
-		logp.Error(err)
+		ec2Logger.Error(err.Error())
+	}
+
+	//Calculate duration based on period
+	durationString, periodSec, err := convertPeriodToDuration(moduleConfig.Period)
+	if err != nil {
+		ec2Logger.Error(err.Error())
```

Comment:
Do we continue on error?

Reply:
`2019-01-23T07:42:14.320-0700    ERROR   instance/beat.go:908    Exiting: 1 error: time: unknown unit ss in duration 300ss accessing '0.period' (source:'/Users/kaiyansheng/go/src/github.com/elastic/beats/x-pack/metricbeat/modules.d/aws.yml')` I was relying on `beat.go` which will exit if period is invalid. But I should add a `return` here anyway to make sure it doesn't continue on error.